src/main/java/com/softserveinc/dokazovi/controller/AuthorController.java

```diff
@@ -0,0 +1,100 @@
+package com.softserveinc.dokazovi.controller;
+
+
+import com.softserveinc.dokazovi.annotations.ApiPageable;
+import com.softserveinc.dokazovi.dto.author.AuthorDTO;
+import com.softserveinc.dokazovi.dto.author.AuthorForAdminDTO;
+import com.softserveinc.dokazovi.dto.payload.ApiResponseMessage;
+import com.softserveinc.dokazovi.security.UserPrincipal;
+import com.softserveinc.dokazovi.service.AuthorService;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.Authorization;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+import static com.softserveinc.dokazovi.controller.EndPoints.AUTHOR_GET_DOCTOR_BY_ID;
+import static com.softserveinc.dokazovi.controller.EndPoints.DOCTOR;
+import static com.softserveinc.dokazovi.controller.EndPoints.DOCTORS;
```

Comment:
Not sure if we actually need both endpoints.

1. Root path
- For `UserController` we have `/user`
![image](https://user-images.githubusercontent.com/86650723/185251618-d660b733-cddb-45b5-b11b-f8d1eb975500.png)
- for `PostController` it is `/post`
![image](https://user-images.githubusercontent.com/86650723/185251788-a22f9af0-8d9e-4384-813b-29a8b8cb090c.png)
Therefore, we should apply `/doctor` for root path here.

2. Post and Put mappings can be applied to the root path as it's done for `PostController`
![image](https://user-images.githubusercontent.com/86650723/185252453-9e82167f-eb76-4c17-a345-f69c756fd7e9.png)

Unless we have other substantial grounds to keep them both.

Reply:
done